projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/DeployJob.tsx

```diff
@@ -0,0 +1,148 @@
+                </div>
+                <div>
+                        <input type="checkbox" name="enable" id="enable" className='jp-vdk-checkbox' onClick={this.onEnableClick()} />
+                        <label className="checkboxLabel" htmlFor="enable">Enable - un-pause the job</label>
+                </div>
+            </>
+        );
+    }
+    /**
+   * Callback invoked upon choosing Enable checkbox
+   */
+    private onEnableClick() {
+        return (event: React.MouseEvent) => {
+            let checkbox = document.getElementById("enable");
+            if (checkbox?.classList.contains("checked")) {
+                checkbox.classList.remove("checked");
+                jobData.set(VdkOption.DEPLOY_ENABLE, "");
+            }
+            else {
+                checkbox?.classList.add("checked");
+                jobData.set(VdkOption.DEPLOY_ENABLE, "1");
+            }
+        }
+    }
+    /**
+  * Callback invoked upon changing the job name input.
+  *
+  * @param event - event object
+  */
+    private onNameChange = (event: any): void => {
+        const nameInput = event.currentTarget as HTMLInputElement;
+        let value = nameInput.value;
+        if (!value) {
+            value = this.props.jobName;
+        }
+        jobData.set(VdkOption.NAME, value);
+    };
+    /**
+   * Callback invoked upon changing the job team input.
+   *
+   * @param event - event object
+   */
+    private onTeamChange = (event: any): void => {
+        const teamInput = event.currentTarget as HTMLInputElement;
+        let value = teamInput.value;
+        if (!value) {
+            value = "default-team";
+        }
+        jobData.set(VdkOption.TEAM, value);
+    };
+    /**
+   * Callback invoked upon changing the job rest-api-url input.
+   *
+   * @param event - event object
+   */
+    private onRestApiUrlChange = (event: any): void => {
+        const urlInput = event.currentTarget as HTMLInputElement;
+        let value = urlInput.value;
+        if (!value) {
+            value = "default-url";
+        }
+        jobData.set(VdkOption.REST_API_URL, value);
+    };
+    /**
+   * Callback invoked upon changing the job path input.
+   *
+   * @param event - event object
+   */
+    private onPathChange = (event: any): void => {
+        const pathInput = event.currentTarget as HTMLInputElement;
+        let value = pathInput.value;
+        if (!value) {
+            value = this.props.jobPath;
+        }
+        jobData.set(VdkOption.PATH, value);
+    };
+    /**
+  * Callback invoked upon changing the deployent reason input.
+  *
+  * @param event - event object
+  */
+    private onDeploymentReasonChange = (event: any): void => {
+        const pathInput = event.currentTarget as HTMLInputElement;
+        let value = pathInput.value;
+        if (!value) {
+            value = "";
+        }
+        jobData.set(VdkOption.DEPLOYMENT_REASON, value);
+    };
```

Comment:
nit: these tens of lines seem very common to me, may extract common code and reuse?

Reply:
changed